src/components/PostProcessing/index.tsx

```diff
@@ -15,50 +14,40 @@ import classnames, {
   space,
 } from 'classnames/tailwind'
 
-const container = (loading?: boolean) =>
+const container = (loading?: boolean, hidden?: boolean) =>
   classnames(
     display('flex'),
     flexDirection('flex-col'),
     alignItems('items-center'),
     backgroundColor('bg-primary-background'),
     padding('py-6', 'px-9'),
     borderRadius('rounded-2xl'),
-    space(loading ? 'space-y-6' : 'space-y-2')
+    space(loading ? 'space-y-6' : 'space-y-2'),
+    display({ hidden })
   )
 
 export default function () {
-  const { pendingPost, lastProcessedStatusId } = useSnapshot(PostIdsStatuses)
+  const { lastUserPost } = useSnapshot(PostIdsStatuses)
 
-  if (!pendingPost && !lastProcessedStatusId) return null
-
-  const pending = !!pendingPost
-
-  const statusComponent = lastProcessedStatusId ? (
-    <LargeText>Tweet successful</LargeText>
-  ) : (
-    <>
-      <ProcessHeader />
-      <Loading />
-    </>
-  )
+  console.log('lastUserPost', lastUserPost)
 
   return (
-    <div className={container(pending)}>
-      {statusComponent}
-      {lastProcessedStatusId ? (
-        <ViewTweetButton
-          url={`https://twitter.com/SealCredEmail/status/${lastProcessedStatusId}`}
-        />
-      ) : pendingPost ? (
-        <ViewTweetButton
-          url={`/tweets/blockchain#store=${pendingPost.store}&id=${pendingPost.id}`}
-          pending
-          internal
-          onClick={() => {
-            SelectedTypeStore.selectedType = pendingPost.store
-          }}
+    <div
+      className={container(
+        lastUserPost?.status === PostStatus.pending,
+        !lastUserPost
+      )}
+    >
+      {lastUserPost?.status === PostStatus.published ? (
+        <TweetSuccessful tweetId={lastUserPost.tweetId} />
+      ) : lastUserPost?.status === PostStatus.rejected ? (
+        <PostRejected
+          store={lastUserPost.store}
+          blockchainId={lastUserPost.blockchainId}
```

Comment:
May we turn it un a separate component that will handle status and return the right component? Via `switch` for instance or smt like that. Like:
```
case 'pending':
  return <PendingPost pendingPost={lastUserPost} />
case 'rejected':
  return <RejectedPost ../>
case 'publish':
  return <SuccessfulPost ... />
```

Reply:
Yeah, I tried that, but didn't work, now I almost finished refactoring and tried again, looks like working :)